analyses/molecular-subtyping-LGAT/03-subset-cnv-files-LGAT.Rmd

```diff
@@ -0,0 +1,304 @@
+                ALT = ALT) %>%
+  # format with chr to match to cnv
+  dplyr::mutate(chr=paste0("chr",chr)) %>%
+  # make genomic ranges
+  GenomicRanges::makeGRangesFromDataFrame(keep.extra.columns = TRUE,
+                                          starts.in.df.are.0based = FALSE)
+
+# overlap manta and domain
+overlaps_manta <- IRanges::mergeByOverlaps(manta_gr, domain_gr)
+
+# filter for fgfr only overlap
+overlaps_manta_fgfr <- overlaps_manta[which(overlaps_manta$Gene.name=="FGFR1" &
+                # filter for tandem duplication
+                overlaps_manta$ALT=="<DUP:TANDEM>"), ]
+
+overlaps_manta_fgfr_df <- data.frame(
+  biospecimen_id = overlaps_manta_fgfr$Kids.First.Biospecimen.ID.Tumor,
+  ALT = overlaps_manta_fgfr$ALT,
+  NAME = overlaps_manta_fgfr$NAME,
+  hgnc_symbol = overlaps_manta_fgfr$hgnc_symbol,
+  # annnotation for dup tandem
+  FGFR_DUP_TANDEM = "Yes",
+  stringsAsFactors = FALSE
+) %>%
+  dplyr::distinct() 
+
+overlaps_manta_fgfr_df 
+
+```
+4 biospecimens have tandem duplication that overlap tyrosine kinase domain in FGFR1
+
+
+### Distribution of copy number
+CNV that are within FGFR1 gene region only and overlap "Protein tyrosine kinase" domain in FGFR1 in biospecimens
+
+```{r}
+
+# gene region overlap
+fgfr_gene = GenomicRanges::GRanges(seqnames=paste0("chr",domain_gr$chromosome_name),
+          ranges=IRanges::IRanges(start = domain_gr$gene_start, end = domain_gr$gene_end))
+
+# subset cnv+domain grange to be within FGFR1 gene region
+cnv_fgfr_gr <- IRanges::subsetByOverlaps(cnv_gr, fgfr_gene,type="within")
+
+# overlap consensus seg with exon
+overlaps_fgfr_cnv <- IRanges::mergeByOverlaps(cnv_fgfr_gr,domain_gr) 
+
+# filter for fgfr only overlap
+overlaps_fgfr_cnv <- overlaps_fgfr_cnv[which(overlaps_fgfr_cnv$hgnc_symbol=="FGFR1" &
+               # filter for duplication
+               overlaps_fgfr_cnv$copy_number>2), ]
+
+# dataframe
+overlaps_cnv_df <- data.frame(
+  biospecimen_id = overlaps_fgfr_cnv$ID,
+  copy_number = overlaps_fgfr_cnv$copy_number,
+  NAME = overlaps_fgfr_cnv$NAME,
+  stringsAsFactors = FALSE
+) %>%
+  dplyr::distinct() %>%
+  # keep copy_number > 2
+  dplyr::mutate(FGFR_DUP = if_else(copy_number >2,"Yes","No")) 
+
+overlaps_cnv_df 
+
+```
+5 samples have copy_number changes > 2 that overlap with tyrosine kinase domain
+
+Are there overlap in SV and CNV calls?
+
+```{r}
+intersect(overlaps_manta_fgfr_df$biospecimen_id,overlaps_cnv_df$biospecimen_id)
+```
+Only BS_MW6YMRBJ shows tandem duplication in manta and duplication in consensus seg file that overlaps tyrosine kinase domain in FGFR1
+
+### Checking which exons overlap the above SV and CNV
+
+```{r}
+# Define the annotations for the hg38 genome
+txdb <- GenomicFeatures::makeTxDbFromGFF(
+    file = file.path(root_dir,"data",
+                     "gencode.v27.primary_assembly.annotation.gtf.gz"),
+    format = "gtf"
+  )
+# get exons and exon_id
+tx_exons<-GenomicFeatures::exons(txdb, 
```

Comment:
It looks like you've filtered this to only to this particular FGFR1 transcript, so the name `tx_exons` is a bit misleading since its only got FGFR1 exons for `ENST00000447712.6`.

Reply:
ENST00000447712.6 seems to be the canonical transcript as per UCSC and also has all exons I'd decided to use this ttranscript

Reply:
My point is here, maybe rename this so its clear that its not all `tx_exons`, its actually `fgfr1_exons`.

Reply:
ah ok, will do

Reply:
thanks, updated the variable name